### PR TITLE
Handle unknown state

### PR DESCRIPTION
**Pull request by bramkragten**

### CODE REVIEW

**Review by iantrich — APPROVED** [review]

File: src/panels/config/devices/ha-devices-data-table.ts [hunk 1]
```diff
@@ -145,7 +145,7 @@ export class HaDevicesDataTable extends LitElement {
                 return html`
                   ${name}<br />
                   ${device.area} | ${device.integration}<br />
-                  ${battery
+                  ${battery && !isNaN(battery.state as any)
```
Comment (balloob) [inline src/panels/config/devices/ha-devices-data-table.ts]: `battery.state` is always a string. You will need to parse it first. But I prefer that you compare it against `unavailable` and `unknown` instead

Comment (balloob) [reply src/panels/config/devices/ha-devices-data-table.ts]: Or this:

```suggestion
                  ${battery && !isNaN(Number(battery.state))
```

Comment (bramkragten) [reply src/panels/config/devices/ha-devices-data-table.ts]: `isNaN` does a string to number conversion, but I can also check for `unavailable` and `unknown`

Comment (balloob) [reply src/panels/config/devices/ha-devices-data-table.ts]: in that case it's fine.

**Pull request merged**